Document preliminary module structure

grst opened the pull request:
- [x] Update README to relfect current consensue on module structure
- [x] Remove test datasets subrepo and add test datasets folder.

drpatelh commented on README.md:
```diff
@@ -56,7 +56,19 @@ nextflow run /path/to/pipeline/ -c /path/to/custom_module.conf
+- Software that can be piped together should be added to separate module files unless there is an run-time, storage advantage in implementing in this way e.g. `bwa mem | samtools view` to output BAM instead of SAM
+- Process names should be all uppercase
+- The `publishDirMode` should be configurable
+- Test data is stored within this repo. Re-use generic files from `tests/data` by
```
Brilliant! Thanks @grst! Could we also add in here the requirements to have a `environment.yml` and Docker file if the software is available on Conda.

grst commented:
@ewels, @drpatelh, added the requirements for software version and environment file to the list. 

(and moved the `markdownlint.yml` to the root directory as like that it's automatically recognized when running markdownlint locally)
(sorry, Phil :stuck_out_tongue:)

drpatelh commented:
This is amazing! Thanks @grst 👍
